homeassistant/components/binary_sensor/homematicip_cloud.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for HomematicIP binary sensor.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.homematicip_cloud/
+"""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.homematicip_cloud import (
+    HomematicipGenericDevice, DOMAIN as HOMEMATICIP_CLOUD_DOMAIN,
+    ATTR_HOME_ID)
+
+DEPENDENCIES = ['homematicip_cloud']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_WINDOW_STATE = 'window_state'
+ATTR_EVENT_DELAY = 'event_delay'
+ATTR_MOTION_DETECTED = 'motion_detected'
+ATTR_ILLUMINATION = 'illumination'
+
+HMIP_OPEN = 'open'
+
+STATE_SABOTAGE = 'sabotage'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the HomematicIP binary sensor devices."""
+    from homematicip.device import (ShutterContact, MotionDetectorIndoor)
+
+    if discovery_info is None:
+        return
+    home = hass.data[HOMEMATICIP_CLOUD_DOMAIN][discovery_info[ATTR_HOME_ID]]
+    devices = []
+    for device in home.devices:
+        if isinstance(device, ShutterContact):
+            devices.append(HomematicipShutterContact(home, device))
+        elif isinstance(device, MotionDetectorIndoor):
+            devices.append(HomematicipMotionDetector(home, device))
+
+    if devices:
+        async_add_devices(devices)
+
+
+class HomematicipShutterContact(HomematicipGenericDevice, BinarySensorDevice):
+    """HomematicIP shutter contact."""
+
+    def __init__(self, home, device):
+        """Initialize the shutter contact."""
+        super().__init__(home, device)
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        if self._device.sabotage:
+            return 'mdi:alert'
+        if self._device.windowState is None:
+            return
+        if self._device.windowState.lower() == HMIP_OPEN:
+            return 'mdi:checkbox-marked-circle-outline'
+
+    @property
+    def state(self):
```

Comment:
Binary sensor platforms should not overwrite the state property, only `is_on`. The state is always either true or false.

Reply:
So I can't use in is_on things like:
if self._device.sabotage:
  return STATE_SABOTAGE
I will remove it for this PR... maybe you can guide me to a better solution.

Reply:
No, binary sensors should only have two states that are the opposite of each other. We use device class to differentiate different binary sensors. Door sensors, motion sensors etc.

Reply:
In home assistant we set the binary state to true to signify an alert that the user should be aware of. So eg if there is sabotage, the sensor should be set to on, ie `is_on` should return `True`.

Reply:
```py
def is_on(self):
    return self._device.sabotage
```